### PR TITLE
SALTO-2127 improved system notes query

### PR DESCRIPTION
**Pull request by dantal4**
A better SuiteQL query for system notes - 10 times faster than before.

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ export default class SuiteAppClient {
   public async runSuiteQL(query: string):
     Promise<Record<string, unknown>[] | undefined> {
     log.debug('Running SuiteQL query: %s', query)
-    if (!/ORDER BY .* ASC/.test(query)) {
+    if (!/ORDER BY .* (ASC|DESC)/.test(query)) {
       log.warn(`SuiteQL ${query} does not contain ORDER BY <unique identifier> ASC, which can cause the response to not contain all the results`)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:113]: Please add DESC to the log message.

**Review by omrilit — COMMENTED** [review]
LGTM, please wait for @alonstern approval

File: packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ export default class SuiteAppClient {
   public async runSuiteQL(query: string):
     Promise<Record<string, unknown>[] | undefined> {
     log.debug('Running SuiteQL query: %s', query)
-    if (!/ORDER BY .* ASC/.test(query)) {
+    if (!/ORDER BY .* (ASC|DESC)/.test(query)) {
```
Comment (alonstern) [inline packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:112]: thanks for that 😪

File: packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts [hunk 3]
```diff
@@ -248,12 +249,6 @@ export default class SuiteAppClient {
       prefer: 'transient',
     }
     const response = await this.safeAxiosPost(url.href, { q: query }, headers)
-    log.debug(
```
Comment (alonstern) [inline packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:251]: why is this removed?

File: packages/netsuite-adapter/src/filters/author_information/system_note.ts [hunk 4]
```diff
@@ -56,12 +56,62 @@ Promise<EmployeeResult[]> => {
...
+    .map(toRecordTypeWhereQuery)
+    .join(' OR ')
+  return `SELECT name, recordid, recordtypeid FROM (SELECT name, recordid, recordtypeid, MAX(date) as date FROM systemnote WHERE ${whereQuery} GROUP BY name, recordid, recordtypeid) ORDER BY date DESC`
+}
+
+const buildFieldSystemNotesQuery = (fieldIds: string[]): string => {
+  const whereQuery = fieldIds
+    .map(toFieldWhereQuery)
+    .join(' OR ')
+  return `SELECT name, field, recordid from (SELECT name, field, recordid, MAX(date) AS date FROM (SELECT name, REGEXP_SUBSTR(field, '^(${FOLDER_FIELD_IDENTIFIER}|${FILE_FIELD_IDENTIFIER})') AS field, recordid, date FROM systemnote WHERE ${whereQuery}) GROUP BY name, field, recordid) ORDER BY date DESC`
```
Comment (alonstern) [inline packages/netsuite-adapter/src/filters/author_information/system_note.ts:81]: nit: can you split this to multiple lines to make it more readable?

**Review by alonstern — APPROVED** [review]
LGTM, approved with some nits

File: packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts (hunk 3 above)
Comment (dantal4) [reply packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:251]: I thought that is log is not indicative and just spams the log file.
